cmd/sharded-test-server/main.go

```diff
@@ -236,6 +234,21 @@ func start(proxyFlags, shardFlags []string, logDirPath, workDirPath string, numb
 		return err
 	}
 
+	// Wait for virtual workspaces to be ready
```

Comment:
Nit: neither this block nor the one below it is waiting for anything, got a bit misled by the comment

Reply:
I'm confused - this block is doing `vw.waitForReady` and the one below is doing `shard.WaitForReady` ?

Reply:
I don't think that's correct - the `waitForReady` does block running in the main goroutine, but then a goroutine is spawned to collect any output from `terminatedCh` and push it to `virtualWorkspacesErrCh` (for later consumption in the `select` at the end) - this same pattern used in #2303